osf/models/mixins.py

```diff
@@ -459,3 +465,59 @@ def get_extra_log_params(self, comment):
         """Return extra data to pass as `params` to `Node.add_log` when a new comment is
         created, edited, deleted or restored."""
         return {}
+
+class TaxonomizableMixin(models.Model):
```

Comment:
I know these functions existed before, but I *really* think `get_subjects()` and `set_subjects` are prime candidates for docstrings. Some bits that I found confusing when reading through:

For `set_subjects`:
- The `preprint_subjects` argument is a list of lists of `Subject` `_ids` (not the objects themselves). 
- Does `preprint_subjects` replace the current object's subjects, or add to them? `self.subjects.clear()` makes me think the former but I would've expected this function to do the latter.

For `get_subjects`:
- What exactly is this function returning? A list of lists of dictionaries containing subject `_ids` and `text`? A documented example return value could be extremely useful so devs don't need to parse through nested for loops that makes calls to a property that contains a list comprehension of another property.

Reply:
Docstring added for `set_subjects`. `get_subjects` was only being used in one place, and unnecessarily. Removed and replaced with a `values_list` call in beceb3e0e26563296a114f49b01e0b1d4a76fdc9